parsec/serde/fields.py

```diff
@@ -105,16 +104,16 @@ def enum_field_factory(enum: Type[Enum]) -> Type[BaseEnumField]:
     return type(f"{enum.__name__}Field", (BaseEnumField,), {"ENUM": enum})
 
 
-def bytes_based_field_factory(value_type: Type) -> Type[Field]:
+def bytes_based_field_factory(value_type: Any) -> Type[Field]:
     assert isinstance(value_type, type)
 
-    def _serialize(self, value, attr, obj):
+    def _serialize(self: Any, value: Any, attr: Any, obj: Any) -> bytes | None:
```

Comment:
Same thing according to the `_serialize` docstring we have:
```
Help on function _serialize in module marshmallow.fields:

_serialize(self, value, attr, obj)
    Serializes ``value`` to a basic Python datatype. Noop by default.
    Concrete :class:`Field` classes should implement this method.

    Example: ::

        class TitleCase(Field):
            def _serialize(self, value, attr, obj):
                if not value:
                    return ''
                return unicode(value).title()

    :param value: The value to be serialized.
    :param str attr: The attribute or key on the object to be serialized.
    :param object obj: The object the value was pulled from.
    :raise ValidationError: In case of formatting or validation failure.
    :return: The serialized value
```

Reply:
Done